src/main/java/org/swisspush/redisques/exception/WastefulExceptionFactory.java

```diff
@@ -16,6 +16,10 @@ class WastefulExceptionFactory implements ExceptionFactory {
     WastefulExceptionFactory() {
     }
 
+    public Exception newException(String message, Throwable cause) {
```

Comment:
I haven't tried to understand how it all works, but in this context I would except that we get some HeavyweightException (or maybe StacktraceException) compared to the ThriftyExceptionFactory that would then give as some LightweightException (StacktraceLessException). It would make things easier to understand I believe. But is this then not convenient to work with?